basics/signals/README.md

```diff
@@ -66,78 +66,134 @@ HTML-like syntax with the tag notation `<>`to represent some visual elements
 (note that you have to add a line: `"jsx": "react",` to the
 `tsconfig.json` file). This is a special syntax used by [React](https://reactjs.org/tutorial/tutorial.html).
 
+You can also try the [React Widget example](./../../react/react-widget) for more details.
+
 `button.tsx` contains one major class `ButtonWidget` that extends the
-`ReactWidget` class provided by JupyterLab. `ReactWidget` defines a
-`render()` method that defines some React elements such as a button. This
-is the recommended way to include React component inside the JupyterLab widget
-based UI.
+`ReactWidget` class provided by JupyterLab.
+
+`ReactWidget` defines a `render()` method that defines some React elements such as a button.
+This is the recommended way to include React component inside the JupyterLab widget based UI.
+
+```ts
+// src/button.tsx#L19-L34
+
+protected render(): React.ReactElement<any> {
+  return (
+    <button
+      key="header-thread"
+      className="jp-example-button"
+      onClick={(): void => {
+        this._count = {
+          clickCount: this._count.clickCount + 1
+        };
+        this._stateChanged.emit(this._count);
+      }}
+    >
+      Clickme
+    </button>
+  );
+}
+```
+
+`ButtonWidget` also contain a private attribute `_count` of type `ICount`.
+
+```ts
+// src/button.tsx#L11-L13
+
+protected _count: ICount = {
+  clickCount: 0
+};
+```
 
 `ButtonWidget` further contains a private variable `_stateChanged` of type
-`Signal`. A signal object can be triggered and then emits an actual message.
+`Signal`.
+
+```ts
+// src/button.tsx#L36-L36
+
+private _stateChanged = new Signal<this, ICount>(this);
+```
+
+A signal object can be triggered and then emits an actual message.
```

Comment:
The lab docstrings seem to be using the term "emit a signal" rather than messages, so we might want to stick to the same terminology.

Reply:
Text is update. Thx for review.